### PR TITLE
AWS query Entity fixes

### PR DESCRIPTION
**Pull request by ashwin-patil**
Required items, please complete
   
   Change(s):
   - Added custom logic to extract Account Name along with UPN if present.
   - replaced IP address custom entities

   Reason for Change(s):
   - Improving correlation capabilities across incidents, alerts and raw events.

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

### CODE REVIEW

File: Solutions/Amazon Web Services/Analytic Rules/AWS_PrivilegeEscalationviaCRUDDynamoDB.yaml [hunk 1]
```diff
@@ -25,17 +25,25 @@ query: |
...
-  | distinct TimeGenerated, EventName, PolicyName, SourceIpAddress, UserIdentityArn, UserIdentityUserName
-  | extend UserIdentityUserName = iff(isnotempty(UserIdentityUserName), UserIdentityUserName, tostring(split(UserIdentityArn,'/')[-1]))
-  | extend timestamp = TimeGenerated, IPCustomEntity = SourceIpAddress, AccountCustomEntity = UserIdentityUserName
+  | extend UserIdentityArn = iif(isempty(UserIdentityArn), tostring(parse_json(Resources)[0].ARN), UserIdentityArn)
+  | extend UserName = tostring(split(UserIdentityArn, '/')[-1])
+  | extend AccountName = case( UserIdentityPrincipalid == "Anonymous", "Anonymous", isempty(UserIdentityUserName), UserName, UserIdentityUserName)
+  | extend AccountName = iif(AccountName contains "@", tostring(split(AccountName, '@', 0)[0]), AccountName),
+    AccountUPNSuffix = iif(AccountName contains "@", tostring(split(AccountName, '@', 1)[0]), "")
+  | distinct TimeGenerated, EventName, PolicyName, SourceIpAddress, RecipientAccountId, AccountName, AccountUPNSuffix, UserIdentityArn, UserIdentityUserName
+  | extend timestamp = TimeGenerated, AccountCustomEntity = UserIdentityUserName
```
Comment (shainw) [inline Solutions/Amazon Web Services/Analytic Rules/AWS_PrivilegeEscalationviaCRUDDynamoDB.yaml:34]: Remove AccountCustomEntity

**Review by shainw — CHANGES_REQUESTED** [review]

**Comment by v-rbajaj** [comment]
There is a mail trail on going for the validation failure for this PR.

**Comment by shainw** [comment]
This will be pending for a couple weeks while the PG implements the entity type we need here.  Do not close.

**Comment by v-rbajaj** [comment]
Hi @shainw, thanks for the update, noted.

**Comment by v-rbajaj** [comment]
Hi @ashwin-patil, please update the branch with master and resolve the merge conflicts.

**Comment by ashwin-patil** [comment]
> Hi @ashwin-patil, please update the branch with master and resolve the merge conflicts.

Done !! Please review and then merge. Thanks .

**Review by v-rbajaj — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Comment by v-rbajaj** [comment]
Hi @shainw, can you please provide your approval here?

**Review by shainw — APPROVED** [review]

**Pull request merged**